src/scales/v.ts

```diff
@@ -0,0 +1,95 @@
+import GradeScale, { findScoreRange, getAvgScore, GradeScales, Tuple } from '../GradeScale'
+import boulder from '../data/boulder.json'
+import { Boulder } from '.'
+export type VScaleType = 'vScale'
+
+const vGradeRegex = /^(V[0-9]{1,2})([/+])?([/-])?([0-9]{1,2})?$/i
+const vGradeIrregular = /^V-([a-zA-Z]*)$/i
+
+const VScale: GradeScale = {
+  displayName: 'V Scale',
+  name: GradeScales.VScale,
+  offset: 1000,
+  isType: (grade: string): boolean => {
+    const isVGrade = (grade.match(vGradeRegex) !== null) || grade.match(vGradeIrregular)
+    // If there isn't a match sort it to the bottom
+    if ((isVGrade === null)) {
+      return false
+    }
+    return true
+  },
+  getScore: (grade: string): number | Tuple => {
+    const parse = grade.match(vGradeRegex)
+    if (parse == null) {
+      // not a valid V scale
+      return 0
+    }
+    const [wholeMatch, basicGrade, plus, dash, secondGrade] = parse
+
+    const basicScore = findScoreRange(
+      (r: Boulder) => r.v.toLowerCase() === basicGrade.toLowerCase(),
+      boulder
+    )
+    // ugly but working will fix later (says ever dev ever)
+    if (wholeMatch !== basicGrade) {
```

Comment:
Idea for future enhancement:  How about strict mode and relaxed mode?  In strict mode we should reject dash-grades eg 5.10a-b, v2-3 or whichever scheme we wish to enforce.